### PR TITLE
SLCORE-412 Binding suggestion feature

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: spec/autobinding.adoc [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+- `sonar.host.url`
+- `sonar.organization`
+
+3. For each filename + tuple of properties, try to guess if we are looking for a SonarQube or SonarCloud connection, using the following heuristic:
+- if the file is `.sonarcloud.properties`, this is SonarCloud (AutoScan)
+- if there is a `sonar.organization`, this is SonarCloud
+- if the `sonar.host.url` is equal to `https://sonarcloud.io` or one of the aliases used for tests, this is SonarCloud, any other non-empty value means it is SonarQube
+- at this stage keep properties as a generic binding clue
+
+At the end of this step, we should have a non-empty list of binding clues having one of those types and attributes:
```
Comment (damien-urruty-sonarsource) [inline spec/autobinding.adoc:45]: Could be empty I think

**Review by damien-urruty-sonarsource — APPROVED** [review]
As discussed I have some doubts & suggestions around names, we can still adjust later

File: spec/autobinding.adoc (hunk 1 above)
Comment (henryju) [reply spec/autobinding.adoc]: Pushed a fix with other improvements

File: client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/config/analysis/AnalysisConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.clientapi.config.analysis;
+
+public class AnalysisConfiguration {
```
Comment (damien-urruty-sonarsource) [inline client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/config/analysis/AnalysisConfiguration.java:22]: This could be removed I think

**Pull request merged**